server/block/item_frame.go

```diff
@@ -0,0 +1,166 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+)
+
+// ItemFrame is a block entity that displays the item or block that is inside it.
+type ItemFrame struct {
+	empty
+	transparent
+
+	// Facing is the direction from the frame to the block.
+	Facing cube.Face
+	// Item is the item that is displayed inside the frame.
+	Item item.Stack
+	// Rotations is the number of rotations for the item in the frame. Each rotation is 45 degrees, with the exception
+	// being maps having 90 degree rotations.
+	Rotations int
+	// Drops is the chance of the item dropping when the frame is broken. In vanilla, this is always 1.0.
+	Drops float64
+	// Glowing makes the frame the glowing variant.
+	Glowing bool
+	// Large makes the frame show up large. This typically only occurs with maps.
+	Large bool
```

Comment:
Does this field actually work with non-map items?

Reply:
No

Reply:
Maybe in that case it'd be better not to expose this as a field but base it on whether the item is a map or not? We obviously don't have maps right now but we could add a TODO for that.